<launch>
    <!-- Argumentos para conectar al robot real -->
    <arg name="robot_ip" default="10.42.0.1"/> <!-- IP del robot en la red -->
    <node pkg="tf" type="static_transform_publisher" name="base_laser_to_laser_tf" args="0 0 0 0 0 1 0 base_laser_link laser 100" />

    <!-- Gmapping Node -->
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
        <remap from="scan" to="/scan"/>
        <param name="odom_frame" value="odom"/>
        <param name="map_frame" value="map"/>
        <param name="base_frame" value="base_link"/>
        <param name="delta" value="0.05"/>
        <param name="linearUpdate" value="0.05"/>
        <param name="angularUpdate" value="0.1"/>
        <param name="xmin" value="-10.0"/>
        <param name="ymin" value="-10.0"/>
        <param name="xmax" value="10.0"/>
        <param name="ymax" value="10.0"/>
        <param name="maxUrange" value="5.6"/>
        <param name="maxRange" value="10.0"/>
        <param name="particles" value="50"/>
        <param name="iterations" value="5"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="map_update_interval" value="0.1"/>
        <param name="transform_publish_period" value="0.05"/>
        <param name="queue_size" value="100"/>
        <param name="transform_tolerance" value="0.2"/>
    </node>

    <!-- RViz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d /home/inaki/catkin_ws/src/arrow_autonomous_navigation_puzzlebot/configs/mapping_real_robot.rviz" />

    <!-- Custom Python Nodes -->
    <node name="Joint_State_Publisher" pkg="arrow_autonomusnavigation_puzzlebot" type="Joint_State_Publisher.py" output="screen"/>
    <node name="localisation" pkg="arrow_autonomusnavigation_puzzlebot" type="localisation.py" output="screen"/>
    <node name="Cordinate_Transform" pkg="arrow_autonomusnavigation_puzzlebot" type="Cordinate_Transform.py" output="screen"/>
    
    <!-- Keyboard Control Python Node -->
    <node name="keyboard_control" pkg="arrow_autonomusnavigation_puzzlebot" type="keyboard_control.py" output="screen"/>

    <!-- PID Parameters -->
    <node name="puzzlebot_controller" pkg="puzzlebot_control" type="puzzlebot_control_node"/>

    <include file="$(find puzzlebot_control)/launch/puzzlebot_control.launch"/>

    <!-- Parámetros de tiempo de simulación -->
    <param name="/use_sim_time" value="false"/>

    <!-- Publicar descripción del robot desde la laptop -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find arrow_autonomusnavigation_puzzlebot)/urdf/puzzlebot_lidar.xacro'"/>

    <!-- Robot State Publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find arrow_autonomusnavigation_puzzlebot)/urdf/puzzlebot_lidar.xacro'"/>
    </node>


</launch>
